presto-spi/src/main/java/com/facebook/presto/spi/type/Chars.java

```diff
@@ -103,16 +103,28 @@ public static Slice trimTrailingSpaces(Slice slice)
         return slice.slice(0, sliceLengthWithoutTrailingSpaces(slice));
     }
 
-    private static int sliceLengthWithoutTrailingSpaces(Slice slice)
+    public static int sliceLengthWithoutTrailingSpaces(Slice slice, int offset, int length)
     {
-        for (int i = slice.length(); i > 0; --i) {
+        requireNonNull(slice, "slice is null");
+        if (length < 0) {
+            throw new IllegalArgumentException("length must be greater than or equal to zero");
+        }
+        if (offset < 0 || offset + length > slice.length()) {
+            throw new IllegalArgumentException("invalid offset/length");
+        }
+        for (int i = length + offset; i > offset; i--) {
             if (slice.getByte(i - 1) != ' ') {
-                return i;
+                return i - offset;
             }
         }
         return 0;
     }
 
+    private static int sliceLengthWithoutTrailingSpaces(Slice slice)
```

Comment:
this is only used from `trimTrailingSpaces()`, so you can inline the call into that method as it's already simple.